velox/exec/tests/HashJoinTest.cpp

```diff
@@ -4252,4 +4254,60 @@ DEBUG_ONLY_TEST_F(HashJoinTest, buildReservationReleaseCheck) {
   VELOX_ASSERT_THROW(runTask(), "");
   ASSERT_TRUE(waitForTaskAborted(task, 5'000'000));
 }
+
+TEST_F(HashJoinTest, joinWithPartitionKey) {
+  parquet::registerParquetReaderFactory(parquet::ParquetReaderType::NATIVE);
+  // bigint.parquet holds one columns (id:BIGINT) and 1 row [0].
+  auto filePath = facebook::velox::test::getDataFilePath(
+      "velox/exec/tests", "data/bigint.parquet");
+  auto split = facebook::velox::exec::test::HiveConnectorSplitBuilder(filePath)
+                   .fileFormat(dwio::common::FileFormat::PARQUET)
+                   .start(0)
+                   .length(472)
+                   .partitionKey("k", "0")
+                   .build();
+  auto outputType = ROW({"n1_0", "n1_1"}, {BIGINT(), BIGINT()});
+  std::unordered_map<std::string, std::shared_ptr<connector::ColumnHandle>>
+      assignments;
+  std::shared_ptr<connector::hive::HiveTableHandle> tableHandle =
+      std::make_shared<connector::hive::HiveTableHandle>(
+          "test-hive",
+          "hive_table",
+          true,
+          connector::hive::SubfieldFilters{},
+          nullptr);
+  assignments["n1_0"] = std::make_shared<connector::hive::HiveColumnHandle>(
+      "id", connector::hive::HiveColumnHandle::ColumnType::kRegular, BIGINT());
+  assignments["n1_1"] = std::make_shared<connector::hive::HiveColumnHandle>(
+      "k",
+      connector::hive::HiveColumnHandle::ColumnType::kPartitionKey,
+      BIGINT());
+  std::vector<RowVectorPtr> buildVectors{
+      makeRowVector({"c0"}, {makeFlatVector<int64_t>({0, 1, 2})})};
+  createDuckDbTable("t", buildVectors);
+  core::PlanNodeId probeScanId;
+  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+  auto op =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(outputType, tableHandle, assignments)
+          .capturePlanNodeId(probeScanId)
+          .hashJoin(
+              {"n1_1"},
+              {"c0"},
+              PlanBuilder(planNodeIdGenerator).values(buildVectors).planNode(),
+              "",
+              {"c0", "n1_0", "n1_1"},
+              core::JoinType::kInner)
+          .project({"c0"})
+          .planNode();
+
+  SplitInput splits = {{probeScanId, {exec::Split(split)}}};
+
+  HashJoinBuilder(*pool_, duckDbQueryRunner_, driverExecutor_.get())
```

Comment:
Should we put some simple validation for the result?

Reply:
Sorry, I didn't quite understand your comments.  Could you give me a reference?
In my opinion, The result of this plan  is "0";
left table has one column c0 [0, 1, 2]
right table is id [0], k[0].
So I use ```select * from t where t.c0 = 0``` to verify the result.